tests/run-tests

```diff
@@ -87,28 +87,28 @@ def run_micropython(pyb, args, test_file, is_special=False):
                     def get(required=False):
                         rv = b''
                         while True:
-                            ready = select.select([master], [], [], 0.02)
-                            if ready[0] == [master]:
-                                rv += os.read(master, 1024)
+                            ready = select.select([primary], [], [], 0.02)
+                            if ready[0] == [primary]:
+                                rv += os.read(primary, 1024)
                             else:
                                 if not required or rv:
                                     return rv
 
                     def send_get(what):
-                        os.write(master, what)
+                        os.write(primary, what)
                         return get()
 
                     with open(test_file, 'rb') as f:
                         # instead of: output_mupy = subprocess.check_output(args, stdin=f)
-                        master, slave = pty.openpty()
-                        p = subprocess.Popen(args, stdin=slave, stdout=slave,
+                        primary, secondary = pty.openpty()
```

Comment:
What do these actually mean? Are there better names than these?

Reply:
A pty is a pseudoterminal, a kind of virtual serial port.  You need it in specific cases where a (sub)program does certain interactive things like provide rich line editing.  There are two ends to these, traditionally called m----- and s----.  In general m----- is the side used by the "controlling" code and s---- is the side used by the "controlled" code.  

In Python, they renamed these in 2018, adopting the "parent" and "child" nomenclature for documentation purposes.   I could mirror that here.  https://github.com/python/cpython/pull/9100

Reply:
Sadly I now see that in 2018 they turned down this change :(

Reply:
Ya, a better response is that we should change unix.

Reply:
None of these terms make sense to me. As someone who hasn't used pseudoterminals, they don't make it clear how they are used. Would `emulator, subterminal` be clearer?

Reply:
https://github.com/creack/pty/commit/7dc38fb350b1d71383eed149e73acb7bae231ddb (openpts for the go language) chose pty/tty, which are related to how the device nodes were named on old unix system.